tycho-plugins/target-platform-utils/src/main/java/org/jboss/tools/tycho/targets/FixVersionsMojo.java

```diff
@@ -112,13 +112,15 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 						fos.write(message.getBytes());
 						fos.write('\n');
 					}
-					if (unit instanceof TargetDefinitionFile.Unit) {
+					// no more possible to set versions in Units. need to parse and set manually by hand
+					//if (unit instanceof TargetDefinitionFile.Unit) {
 						// That's deprecated, but so cool (and no other way to do it except doing parsing by hand)
-						((TargetDefinitionFile.Unit)unit).setVersion(version);
-					}
+						//((TargetDefinitionFile.Unit)unit).setVersion(version);
+					//}
 				}
 			}
-			TargetDefinitionFile.write(targetDef, new File(targetFile.getParent(), targetFile.getName() + "_fixedVersion.target"));
+			// since no automatic modifications, there is no need to create a new target platform.
+			//TargetDefinitionFile.write(targetDef, new File(targetFile.getParent(), targetFile.getName() + "_fixedVersion.target"));
 		} catch (FileNotFoundException ex) {
```

Comment:
@jeffmaury question here : as it's not possible to fix versions, should we deprecate this Mojo entirely or provide the information as a warning in console log ?

Reply:
We need to find the alternative way of fixing the version as this is used during our TP update releng, no ?